### ISSUE TITLE
cannot read property fd of undefined

### ISSUE THREAD

**Issue by ronag** [pinojs/pino#131]
```
foo/node_modules/pino/pino.js:123
        var fd = (istream.fd) ? istream.fd : istream._handle.fd
                                                            ^

TypeError: Cannot read property 'fd' of undefined
    at foo/node_modules/pino/pino.js:123:61
    at f (foo/node_modules/once/once.js:25:25)
    at process.onExit.insertCode (foo/node_modules/pino/pino.js:444:7)
    at emitOne (events.js:96:13)
    at process.emit (events.js:188:7)
    at processEmit [as emit] (foo/node_modules/fsevents/node_modules/signal-exit/index.js:146:32)
    at process._fatalException (bootstrap_node.js:284:26)
```

```javascript
var pino = require('pino')
var pretty = pino.pretty()
pretty.pipe(process.stdout)
var log = pino({
  name: 'app',
  safe: true
}, pretty)

log.child({ widget: 'foo' }).info('hello')
log.child({ widget: 'bar' }).warn('hello 2')
```

Node v7

**Comment by jsumners** [comment]
This should only be possible if you are using "extreme" mode. Indeed, using your test script, I do not see the issue:

```bash
% node test.js                                                                                                                                     
[2016-11-23T13:30:22.891Z] INFO (app/59494 on sumners-osx.int.clayton.edu): hello
    widget: "foo"
[2016-11-23T13:30:22.897Z] WARN (app/59494 on sumners-osx.int.clayton.edu): hello 2
    widget: "bar"

[jsumners@sumners-osx] /tmp/23
% node --version                                                                                                                                   
v7.2.0
```

**Comment by davidmarkclements** [comment]
sidepoint - `safe` is true by default - @ronag could you give us exact node version, also what OS are you on?

**Comment by ronag** [comment]
```javascript
const pretty = pino.pretty()
pretty.pipe(process.stdout)
const logger = pino({
  name: 'nxt-file',
  extreme: true
}, pretty)
```

OSX 10.12, v7.0.0

**Comment by mcollina** [comment]
which version of pino? We should have adressed that in the past. You can't use extreme mode with anything that is not a socket or a file stream.

**Comment by mcollina** [comment]
(you should see a better error than this one)

**Comment by ronag** [comment]
3.0.1

Could we have a wrapper around the destination if it does not support extreme mode?

**Comment by jsumners** [comment]
It's odd. The pretty stream is passing the `isBlockableStream` test, but failing once it gets to clearing the buffer. It either should fail the initial test or succeed when clearing the buffer.

**Comment by jsumners** [comment]
Okay, this is actually covered in the documentation as the last bullet point in "Extreme Mode Explained" -- https://github.com/pinojs/pino#extreme-mode-explained

This test script "works":

```js
'use strict'

const pino = require('pino')
const pretty = pino.pretty()
pretty.pipe(process.stdout)

const log = pino({extreme: true}, pretty)
log.on('error', function (err) {
  console.log('cannot start in exreme mode: ' + err.message)
  process.exit(1)
})

log.info('foo')
```

Note that the failing code will also fail on Node 6.9.1; it's just plain wrong.

However, I think this issue is bringing something to light: pino needs a 'ready' event for extreme mode. This is the scenario I tried to avoid with my original synchronous init for the blocking stream test.

**Comment by mcollina** [comment]
Just to clarify: using extreme mode for anything that does not do direct I/O is wrong. Extreme mode speeds up direct I/O. If it needs to pass through a transform, it will loose all its benefits.

**Comment by mcollina** [comment]
@jsumners is this covered in #92? Or can you assemble a PR to make sure that it does not crash with an unexpected and unknown error message? I'm ok in crashing with _our own_ error message, pointing to some docs that tells _not_ to do that.

**Comment by jsumners** [comment]
@mcollina I suppose it could be related to #92, but I'm not entirely convinced  of it.

**Comment by davidmarkclements** [comment]
what do we need to do to close this out?

**Comment by jsumners** [comment]
@davidmarkclements close #92.

**Comment by jsumners** [comment]
Fixed by #187
